src/Gui/propertyeditor/PropertyItem.h

```diff
@@ -1,1272 +1,1329 @@
- */
-class GuiExport PropertyUnitConstraintItem: public PropertyUnitItem
-{
-    Q_OBJECT
-    PROPERTYITEM_HEADER
-
-    void setEditorData(QWidget *editor, const QVariant& data) const override;
-
-    void setRange(double min, double max)
-    {
-        this->min = min;
-        this->max = max;
-    }
-
-    void setStepSize(double steps)
-    {
-        this->steps = steps;
-    }
-
-protected:
-    PropertyUnitConstraintItem();
-
-private:
-    double min = double(INT_MIN);
-    double max = double(INT_MAX);
-    double steps = 0.1;
-};
-
-/**
- * Change a floating point number with constraints.
- * \author Werner Mayer
- */
-class GuiExport PropertyFloatConstraintItem: public PropertyItem
-{
-    Q_OBJECT
-    PROPERTYITEM_HEADER
-
-    QWidget* createEditor(QWidget* parent, const std::function<void()>& method) const override;
-    void setEditorData(QWidget *editor, const QVariant& data) const override;
-    QVariant editorData(QWidget *editor) const override;
-
-    void setRange(double min, double max)
-    {
-        this->min = min;
-        this->max = max;
-    }
-
-    void setStepSize(double steps)
-    {
-        this->steps = steps;
-    }
-
-protected:
-    QVariant toString(const QVariant&) const override;
-    QVariant value(const App::Property*) const override;
-    void setValue(const QVariant&) override;
-
-protected:
-    PropertyFloatConstraintItem();
-
-private:
-    double min = double(INT_MIN);
-    double max = double(INT_MAX);
-    double steps = 0.1;
-};
-
-/**
- * Change a floating point number with many decimal points (hard coded as 16)
- */
-class GuiExport PropertyPrecisionItem: public PropertyFloatConstraintItem
-{
-    Q_OBJECT
-    PROPERTYITEM_HEADER
-protected:
-    PropertyPrecisionItem();
-};
-
-/**
- * Change a floating point number.
- * \author Werner Mayer
- */
-class GuiExport PropertyAngleItem : public PropertyUnitConstraintItem
-{
-    Q_OBJECT
-    PROPERTYITEM_HEADER
-
-protected:
-    void setEditorData(QWidget *editor, const QVariant& data) const override;
-    QVariant toString(const QVariant&) const override;
-
-protected:
-    PropertyAngleItem();
-};
-
-/**
- * Edit properties of boolean type. 
- * \author Werner Mayer
- */
-class GuiExport PropertyBoolItem: public PropertyItem
-{
-    Q_OBJECT
-    PROPERTYITEM_HEADER
-
-    QWidget* createEditor(QWidget* parent, const std::function<void()>& method) const override;
-    void setEditorData(QWidget *editor, const QVariant& data) const override;
-    QVariant editorData(QWidget *editor) const override;
-
-protected:
-    QVariant value(const App::Property*) const override;
-    void setValue(const QVariant&) override;
-
-protected:
-    PropertyBoolItem();
-};
-
-/**
- * Edit properties of vector type. 
- * \author Werner Mayer
- */
-class PropertyFloatItem;
-class GuiExport PropertyVectorItem: public PropertyItem
-{
-    Q_OBJECT
-    Q_PROPERTY(double x READ x WRITE setX DESIGNABLE true USER true) // clazy:exclude=qproperty-without-notify

```

Comment:
All the automatic line breaks of Q_PROPERTY caused by clang-format should be explicitly disabled with `// clang-format off` and `// clang-format on`. This is because the code looks worse and I am not sure if suppressing the clazy warnings this way still works.

Reply:
Done. Please note this part has separate PR #15970, so once agreed upon, it could be merged without waiting for the rest of units clean up is sorted.